x/poolmanager/keeper/create_pool.go

```diff
@@ -60,15 +60,24 @@ func (k Keeper) CreatePool(ctx sdk.Context, msg types.CreatePoolMsg) (uint64, er
 		return 0, fmt.Errorf("cannot create module account %s, "+
 			"due to an account at that address already exists", poolAddr)
 	}
-	k.accountKeeper.SetModuleAccount(ctx, authtypes.NewEmptyModuleAccount(poolName))
+	// Create a new module account for the pool.
+	moduleAccount := authtypes.NewEmptyModuleAccount(poolName)
+	moduleAccountI, ok := (k.accountKeeper.NewAccount(ctx, moduleAccount)).(authtypes.ModuleAccountI) // set the account number
+	if !ok {
+		return 0, fmt.Errorf("cannot create module account %s, "+
+			"failed to set account number", poolAddr)
+	}
+	k.accountKeeper.SetModuleAccount(ctx, moduleAccountI)
```

Comment:
```suggestion
	moduleAccount := authtypes.NewEmptyModuleAccount(poolName)
	k.accountKeeper.SetModuleAccount(ctx, moduleAccount)
```

Reply:
@zale144 we need to set it that way to prevent it from having 0 as account number. I discussed it with @mtsitrin

Reply:
@hoangdv2429 just saw your reply. Got it, I updated my comment with another way it can be done

Reply:
@hoangdv2429 @mtsitrin this way no need for error check

Reply:
not sure what the gain here.
I prefer the original way, as used in the vanilla SDK `GetModuleAccountAndPermissions`
we don't use it as it expects the module accounts to be registered in `app.go`.
as we create module accounts on the run, we just copy paste the relevant snippet

Reply:
I see. Then best keep the code consistent. I just thought it would be a bit more concise and avoid err check, I didn't know how it was exactly done in SDK